File: Array/560-subarray-sum-equals-k/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> prefixsumcount;
        int count=0;
        int sum=0;
        prefixsumcount[0]=1;
        for(auto num:nums){
            sum+=num;
            if(prefixsumcount.find(sum-k)!= prefixsumcount.end()){
               count+= prefixsumcount[sum-k];
            }
            prefixsumcount[sum]++;
        }
        return count;
        
    }
};

File: Linked_List/560-subarray-sum-equals-k/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {

        int count=0;


        int n = nums.length;


        for(int i=0;i<n;i++){

            int j=i+1;
            int sum = nums[i];

            if(sum==k) count++;

            while(j<n){

                sum+=nums[j];

                if(sum==k) count++;
                j++;
            }
        }
        return count;
        
    }
}